### ISSUE TITLE
Why disable the Gradle daemon

### ISSUE THREAD

**Issue by hmijail** [circleci/circleci-docs#4038]
# Locations
https://circleci.com/docs/2.0/language-java/

# Details
This example disables the Gradle daemon with no explanation. But the Gradle docs recommend to enable it everywhere.
https://docs.gradle.org/current/userguide/gradle_daemon.html

One of the Gradle devs emphasized use of the daemon on CI:
https://blog.andresteingress.com/2018/09/05/gradle-daemon-and-jenkins.html

If there is an actual reason to disable the daemon, please state it.

**Comment by rosieyohannan** [comment]
@hmijail apologies for the delay with this. I'm in touch with the engineers that wrote the code example so will be back soon with either a fix or an explanation ☺️ .

**Comment by lokst** [comment]
@hmijail, our example disables the Gradle daemon to let the Gradle process terminate after it is done. This helps to conserve memory and reduce the chance of OOM errors.

**Comment by rosieyohannan** [comment]
Fixed in #4201

**Comment by hmijail** [comment]
Sorry for the late answer, I was on holidays.

I'm struggling to see how that explanation makes sense. Docker containers exit when the process that was started as an entry point exits; any daemon will be killed automatically.

Demonstration:
```
# daemonizing a "sleep 10" command and running "ls" finishes as soon as "ls" does: the daemon doesn't matter
$ /usr/bin/time docker run -it circleci/openjdk /bin/bash -c "sleep 10 & ls>/dev/null"
        0.82 real         0.03 user         0.07 sys

# but if "sleep" runs in the foreground, the container stays alive as long as the "sleep" 
$ /usr/bin/time docker run -it circleci/openjdk /bin/bash -c "sleep 10"
       10.77 real         0.03 user         0.07 sys
```

More concretely, I am currently running my gradle tasks with the daemon **enabled** and they work as expected.
